### DISCUSSION TITLE
support plausible.io integration

### DISCUSSION THREAD

**Discussion by stefan-berkner-tilotech** [retypeapp/retype#272] [Ideas]
We use plausible.io as an alternative to google analytics due to its privacy approach. However, we would need to add the following html tag into our generated docs:

```javascript
<script defer data-domain="example.com" src="https://plausible.io/js/plausible.js"></script>
```

I tried various approaches, including to put that somehow in the footer, but nothing worked. I guess the only way to get it working would be to officially support it similar to the [google analytics integration](https://retype.com/configuration/project/#googleanalytics).

A more generic approach would of course be the support for javascript tags.

**Comment by DannyBen** [comment]
I think this is a great idea. I also like plausible.io, and think that both implementations (support additional analytics AND support custom injectables) would be a great addition to Retype.

**Comment by geoffreymcgill** [comment]
We might be able to integrate Plausible. Configuration looks like it would only need to a simple `enabled` boolean with default being `false`. If you want to enable Plausible, set `integrations.plausible.enabled: true` in your projects `retype.yml` configuration file. 

```yml
integrations:
  plausible:
    enabled: true | false (default is false)
```

Retype will use your `url` config to set the `data-domain` attribute. The following `<script>` would be emitted into the `<head>` of all pages:

```html
<script defer data-domain="{{ url }}" src="https://plausible.io/js/plausible.js"></script>
```
How does that sound?

**Reply by DannyBen** [reply]
Sounds super!

**Reply by stefan-berkner-tilotech** [reply]
Thanks for showing interest for the proposal.
I guess the domain should optionally also be configurable, with the default value being the configured `url`. Reason for that is, that we host the documentation under `docs.example.com` but would want the `data-domain` to be `example.com`.
Also plausible.io supports sending the [stats to multiple domains](https://plausible.io/docs/plausible-script#can-i-send-stats-to-multiple-dashboards-at-the-same-time) like this:
```javascript
<script defer data-domain="domain1.com,domain2.com,subdomain.yourdomain.com" src="https://plausible.io/js/plausible.js"></script>
```

**Comment by geoffreymcgill** [comment]
We went with a `domain` config instead of the `enabled` idea. 

The `domain` config requires a value (is not `null|empty`) for the plausible `<script>` snippet to be added to the website.

```yml
integrations:
  plausible:
    domain: example.com
```

The above configuration outputs the following snippet into the `<head>` of all pages:

```html
<script defer data-domain="example.com" src="https://plausible.io/js/plausible.js"></script>
```

The `domain` also accepts an array of values, such as `example.com,docs.example.com`:

```yml
integrations:
  plausible:
    domain: example.com,docs.example.com
```

The above config outputs the following `<script>`:

```html
<script defer data-domain="example.com,docs.example.com" src="https://plausible.io/js/plausible.js"></script>
```

The `plausible` integration will be available in the upcoming Retype `v2.0` release. No release date has been set for `v2.0` but we're getting close.

Hope this helps.

**Reply by stefan-berkner-tilotech** [reply]
Perfect! Thanks a lot for the fast implementation! Looking forward to v2.0 then.

**Reply by geoffreymcgill** [reply]
Plausible.io integration is now available with Retype [`v2.0.0-preview`](https://github.com/retypeapp/retype/discussions/287#discussioncomment-2321442).

```yml
integrations:
  plausible:
    domain: <domains>
```

A sample configuration would be:

```yml
integrations:
  plausible:
    domain: example.com,docs.example.com
```

We still have the documentation to write but everything should be available before the final `v2.0.0` release which is scheduled for next week. 

Hope this helps.

**Reply by geoffreymcgill** [reply]
Plausible integration docs have been added, see https://retype.com/configuration/project/#plausible.

**Reply by stefan-berkner-tilotech** [reply]
Perfect. We did try and deploy it today. Great job.

**Reply by viewsourc3** [reply]
This is great, thank you <3

**Reply by DannyBen** [reply]
I think there is a missing configuration option.
Plausible is also supported as open source, which means you can host it yourself. In this case, the `src` attribute of the pasted snippet should change to be the self hosted version:

```html
<script defer data-domain="my.retype.site.com" src="https://my.self.hosted.plausible.com/js/plausible.js"></script>
```

I was looking for:

```yaml
integrations:
  plausible:
    domain: bashly.dannyb.co
    host: plausible.dannyb.co   # <== need this
```

In fact - I was sure that the `domain` option is actually setting the plausible host - after all the `domain` is already present in the retype options, in the `url` option...

Can this be supported?
In the meantime I will search for ways to inject this manually.

**Reply by geoffreymcgill** [reply]
We can add `host` as a config option on the `plausible` object. 

As a temp work around, you can add that script to a `_includes/head.html` file.

#### _includes/head.html

```html
<script defer data-domain="my.retype.site.com" src="https://my.self.hosted.plausible.com/js/plausible.js"></script>
```

Once added to `_includes/head.html`, anything within that file will be included on all generated pages of the website.

Hope this helps.

**EDIT**: Additional details on the includes available at https://github.com/retypeapp/retype/discussions/282#discussioncomment-2343005

**Reply by DannyBen** [reply]
Excellent. Thanks for the list - I looked for it in the documentation since I remembered I saw this functionality somewhere.

**Reply by geoffreymcgill** [reply]
The new `host` config option has been implemented and will be available in the next release of Retype.

Documentation available at https://retype.com/configuration/project/#plausiblehost.
